### DISCUSSION TITLE
Is SPI 3-wire supported on rapsberry pi 4b ?

### DISCUSSION THREAD

**Discussion by AlexandreBourrieau** [rsta2/circle#398] [Q&A] [answered]
Hello,
Is the 3-wires SPI is supported using raspberry pi 4b with circle ?
Thanks you !

**Comment by rsta2** [comment]
Yes, the Circle `CSPIMaster[DMA]` drivers implement the 3-wire SPI standard mode with MOSI, MISO, SCLK as described in section 9.2.1 in [this document](https://datasheets.raspberrypi.com/bcm2711/bcm2711-peripherals.pdf).

**Reply by AlexandreBourrieau** [reply]
Thanks you;

I would say 3-wires = CLK, CS and MIMO like in section 9.2.2 = Bidirectional mode.
(in my mind CLK,MOSI,MISO,CLK is 4-wire = standard mode)

Is the library support it or do I need personnaly to access internal register to configure it like seeting the SPI_REN bit in the SPI control & status register SPI_CS before reading data ?

**Reply by rsta2** [reply, marked as answer]
Ah OK. Unfortunately the bidirectional mode as in section 9.2.2 is not supported yet.

**Reply by AlexandreBourrieau** [reply]
Ok thanks you !
